packages/extension-base/src/setup-app.ts

```diff
@@ -17,21 +17,22 @@ import {Config} from './types'
 import withRedBorder from 'overridable!./components/with-red-border'
 import SamplePage from './pages/sample'
 
-const defaultPath = '/sample-page'
 import extensionMeta from '../extension-meta.json'
 
 class Sample extends ApplicationExtension<Config> {
     static readonly id = extensionMeta.id
 
-    extendApp<T>(App: React.ComponentType<T>): React.ComponentType<T> {
+    extendApp<T extends React.ComponentType<T>>(
+        App: React.ComponentType<T>
+    ): React.ComponentType<T> {
```

Comment:
Bonus: fixed this type error by matching the generic type with what's defined in the parent class.